manimlib/utils/tex.py

```diff
@@ -0,0 +1,258 @@
+import os
+from manimlib.utils.config_ops import digest_config
+
+class TexTemplateFromFile():
+    """
+    Class representing a TeX template file
+    """  # TODO: attributes, dataclasses stuff
+    CONFIG = {
+        "use_ctex": False,
+        "filename" : "tex_template.tex",
+        "text_to_replace": "YourTextHere",
+    }
+    template_text = ""
+
+    def __init__(self, **kwargs):
+        digest_config(self, kwargs)
+        self.rebuild_cache()
+        
+    def rebuild_cache(self):
+        """For faster access, the LaTeX template's code is cached. 
+        If the base file is modified, the cache needs to be rebuilt.
+        """
+        with open(self.filename, "r") as infile:
+            self.template_text = infile.read()
+
+    def get_text_for_text_mode(self,expression):
+        """Inserting expression verbatim into TeX template.
+        
+        Parameters
+        ----------
+        expression : :class:`str`
+            String containing the expression to be typeset, e.g. `"foo"`
+        
+        Returns
+        -------
+        :class:`str`
+            LaTeX code based on the template containing the given expression and ready for typesetting.
+        """
+        return self.template_text.replace(
+            self.text_to_replace, expression
+        )
+
+    def get_text_for_tex_mode(self,expression):
+        """Inserts an expression into the TeX template, surrounded
+        by `\\begin{align*} ... \\end{align*}` for math mode.
+        Parameters
+        ----------
+        expression : :class:`str`
+            The string containing the (math) expression to be typeset, e.g. $\\sqrt{2}$
+            
+        Returns
+        -------
+        :class:`str`
+            LaTeX code based on template, containing the given expression and ready for typesetting
+        """
+        return self.template_text.replace(
+            self.text_to_replace,
+            r"\begin{align*}" "\n" \
+            + expression + "\n" \
+            r"\end{align*}"
+        )
```

Comment:
Any specific reason for the particular choice of `align*`? I mean, it's certainly very useful, but we could generalize this for any `\begin{...} ... \end{...}`, no?

```suggestion
    def get_text_for_env(self, environment, expression):
        """Inserts an expression into the TeX template, surrounded
        by `\\begin{<environment>} ... \\end{<environment>}` for a certain environment.
        
        Parameters
        ----------
        environment : :class:`str`
            The environment in which we should wrap the expression.
        expression : :class:`str`
            The string containing the expression to be typeset, e.g. $\\sqrt{2}$
            
        Returns
        -------
        :class:`str`
            LaTeX code based on template, containing the given expression and ready for typesetting
        """
        begin = r"\begin{" + environment + "}"
        end = r"\end{" + environment + "}"
        return self.template_text.replace(
            self.text_to_replace,
            "{0}\n{1}\n{2}".format(begin, expression, end)
        )

    def get_text_for_tex_mode(self,expression):
        """Inserts an expression into the TeX template, surrounded
        by `\\begin{align*} ... \\end{align*}` for math mode.
        
        Parameters
        ----------
        expression : :class:`str`
            The string containing the (math) expression to be typeset, e.g. $\\sqrt{2}$
            
        Returns
        -------
        :class:`str`
            LaTeX code based on template, containing the given expression and ready for typesetting
        """
        return get_text_for_env("align*", expression)
```

Reply:
The reason for this is that Grant used this exact structure (see `constants.py` in its original version) and I wanted to stay as close to his idea as possible. I am more than happy with your solution.